lmfit/confidence.py

```diff
@@ -249,17 +255,12 @@ def calc_ci(self, para, direction):
                 ret.append((prob, direction*np.inf))
                 continue
 
-            try:
-                val = brentq(calc_prob, a_limit,
-                             limit, rtol=.5e-4, args=prob)
-            except ValueError:
-                self.reset_vals()
-                try:
-                    val = brentq(calc_prob, start_val,
-                                 limit, rtol=.5e-4, args=prob)
-                except ValueError:
-                    val = np.nan
-
+            sol = root_scalar(calc_prob, method='toms748', bracket=sorted([limit, a_limit]), rtol=.5e-4, args=(prob,))
+            if sol.converged:
+                val = sol.root
+            else:
+                val = np.nan
+                break
```

Comment:
sorry, it's late so I might be missing something.... why is the `break` needed here? There isn't a loop or anything going on so it seems redundant to me

Reply:
sorry, as Matt pointed out it is actually in a "for-loop" - so my comment is wrong...